packages/nx/src/config/workspaces.ts

```diff
@@ -653,36 +654,12 @@ export function globForProjectFiles(
   const combinedProjectGlobPattern = '{' + projectGlobPatterns.join(',') + '}';
 
   performance.mark('start-glob-for-projects');
-  /**
-   * This configures the files and directories which we always want to ignore as part of file watching
-   * and which we know the location of statically (meaning irrespective of user configuration files).
-   * This has the advantage of being ignored directly within globSync
-   *
-   * Other ignored entries will need to be determined dynamically by reading and evaluating the user's
-   * .gitignore and .nxignore files below.
-   */
-
-  const ALWAYS_IGNORE = [
-    '/node_modules',
-    '**/node_modules',
-    '/dist',
-    ...globsToExclude,
-  ];
 
-  /**
-   * TODO: This utility has been implemented multiple times across the Nx codebase,
-   * discuss whether it should be moved to a shared location.
-   */
   const ig = ignore();
```

Comment:
We don't need to save the ignore object, and can do it outside of deduplicateProjectFiles